OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/UserState.kt

```diff
@@ -0,0 +1,28 @@
+package com.onesignal.user
+
+import org.json.JSONObject
+
+/**
+ * A user state.
+ */
+class UserState(
+    /**
+     * The unique identifier for your OneSignal account. This will be an empty string until the
+     * user has been successfully logged in on the backend and assigned an ID.
+     * Use [addObserver] to be notified when the [onesignalId] has been successfully assigned.
+     */
+    val onesignalId: String,
+    /**
+     * The external identifier that you use to identify users. This will be an empty string
+     * until the user has been successfully logged in on the backend and assigned an ID.
+     * Use [addObserver] to be notified when the [externalId] has
+     * been successfully assigned.
```

Comment:
This sentence "This will be an empty string _until_ the user has been successfully logged in on the backend and assigned an ID" could be confusing for app developers. Perhaps something like:

> The external identifier that you use to identify users. Use [addObserver] to be notified when the [externalId] has been successfully assigned. This will be an empty string if no external identifier has been assigned to the associated [onesignalId].